client/my-sites/site-settings/form-base.js

```diff
@@ -62,6 +62,11 @@ module.exports = {
 
 	},
 
+	recordClickEventAndStop: function( recordObject, clickEvent ) {
+		this.recordEvent( recordObject );
+		clickEvent.preventDefault();
```

Comment:
@artpi I think the intent here is to stop the click event from bubbling up to the label as you noted so we only need `stopPropagation`. Using `preventDefault` here might mislead some one into thinking we want to prevent the default action of the `select` element.